### DISCUSSION TITLE
Welcome to Android-Roadmap Discussions!

### DISCUSSION THREAD

**Discussion by ahmed-shehataa** [Eng-AhmedShehata/Android-Roadmap#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Marwan28** [comment]
Hi, I'm Marwan.
I'm learning android basics now, and i already did my graduation project with it.
Now, i hope to have more experience in this field, so i want to ask if this road map nowadays is the best or their are some improvements on it.
Thanks.

**Reply by ahmed-shehataa** [reply]
Hi marwan,
This road map contains basic, intermediate then advanced topics. 
For advanced topics like:
Dependency injection, reactive programing, services, compose, clean code, mvi architecture and more. 
But for now I didn't provide any resources for all these topics so I hope soon to add resources for them. 
Just try to search and read more about them then practice with a simple project every time.
If you need any help, don't hesitate to contact me 
(Facebook, LinkedIn, twitter ) in my github profile. 
Thanks

**Comment by Marwan28** [comment]
First of all thanks for your response
Second
I want to ask about Flutter
Is that better to learn flutter or continue in android?
Thanks for your time ❤️

**Reply by ahmed-shehataa** [reply]
As you know 
Flutter is a cross platform mobile application framework(android - ios) 
but android development for only android applications.

So in my opinion,
Learning android and its core, basics, threading, services and some advanced topics may make you easy to shift or learning another platform easy with no problems.
You should know that some flutter developers sometimes need to write a native code ( android - ios) for handling some hardware or some app functionality that flutter can't handle it so you need to use something called ( channels ) in flutter to write a native code.

Finally, 
I was in this struggle with myself ( continue android development or shifting to flutter) but finally I choose to continue in android development.
This opinion only belongs to me and not a rule or a standard that you should align 😅. 
So you need to read about both and choose what you want to learn! 👌

What ever you choose enjoy what you learn ( android or flutter) 🥳

**Comment by Marwan28** [comment]
Thanks for the feedback!
I will read more about Flutter and android
